x-pack/plugins/endpoint/server/services/endpoint/alert_query_builders.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { KibanaRequest } from 'kibana/server';
+import { EndpointAppConstants } from '../../../common/types';
+import { EndpointAppContext } from '../../types';
+
+export const kibanaRequestToAlertListQuery = async (
+  request: KibanaRequest<any, any, any>,
+  endpointAppContext: EndpointAppContext
+): Promise<Record<string, any>> => {
+  const pagingProperties = await getPagingProperties(request, endpointAppContext);
+  return {
+    body: {
+      query: {
+        match_all: {},
+      },
+      sort: [
+        {
+          '@timestamp': {
+            order: 'desc',
+          },
+        },
+      ],
+    },
+    from: pagingProperties.pageIndex * pagingProperties.pageSize,
+    size: pagingProperties.pageSize,
+    index: EndpointAppConstants.ALERT_INDEX_NAME,
+  };
+};
+
+async function getPagingProperties(
```

Comment:
Question @nnamdifrankie, I think this originated from your initial PR. I think what this is doing is looping over all the key value pairs in the paging_properties of the request and creating a new object that we then only pull out `page_size` and `page_index`. Is this relating to the bug that you posted or could we instead just do something like:

```
if (request?.body?.page_properties.page_size)
  set page size
if (request?.body?.page_properties.page_index)
  set page_index
```

Reply:
@jonathan-buttner It's due to this bug: https://github.com/elastic/kibana/issues/54843 ... I hope we fix this and make this less awkward. 🙃

Reply:
ah ok